src/explorer/AzureAccountTreeItem.ts

```diff
@@ -67,16 +69,19 @@ export class AzureAccountTreeItem extends AzureAccountTreeItemBase {
     }
 
     private async loadTrialAppNode(): Promise<AzExtTreeItem | undefined> {
-        const loginSession: string | undefined = ext.context.globalState.get(TrialAppLoginSession);
-        if (!loginSession) {
+        const trialAppContext: ITrialAppContext | undefined = ext.context.globalState.get(TrialAppContext);
+        if (!trialAppContext) {
             this.trialAppNode = undefined;
             return undefined;
         }
 
         const ti: AzExtTreeItem[] = await this.createTreeItemsWithErrorHandling(
-            [loginSession],
+            [trialAppContext.loginSession],
             'trialAppInvalid',
             async (source: string): Promise<AzExtTreeItem> => {
+                if (new Date(trialAppContext.expirationDate) < new Date()) {
```

Comment:
I'd use a ternary just so it's clear there's 2 different types of trees that we can return here